lib/tests/test_merge_trees.rs

```diff
@@ -208,6 +208,52 @@ fn test_same_type(use_git: bool) {
     };
 }
 
+#[test_case(false ; "local backend")]
+#[test_case(true ; "git backend")]
+fn test_executable(use_git: bool) {
+    let test_repo = TestRepo::init(use_git);
+    let repo = &test_repo.repo;
+    let store = repo.store();
+
+    // The file name encodes whether the file was executable or normal in the base
+    // and in each side
+    let files = vec!["nnn", "nnx", "nxn", "nxx", "xnn", "xnx", "xxn", "xxx"];
+
+    let write_tree = |index: usize| -> Tree {
```

Comment:
[Optional]: I think this would be more readable if `write_tree` took an iterator of string & bool pairs and there was a separate function 

```rust
|files: Iterator<&str>, index: usize| -> Iterator<(&str, bool)> =
files.filter(|f| f.size() < index).map(|f| (f, f[index..index+1] == 'x')
```

This would also allow examples with more than 3 characters.

Reply:
Sure, done.